h/views/accounts.py

```diff
@@ -572,6 +576,63 @@ def _template_data(self, js_config=None):
         return {"js_config": js_config}
 
 
+@view_config(
+    route_name="account_identity",
+    is_authenticated=True,
+    request_method="POST",
+    require_csrf=True,
+    request_param=("provider", "provider_unique_id"),
+)
+def delete_identity(request):
+    user = request.user
+    db = request.db
+    flash = request.session.flash
+    params = request.params
+    route_url = request.route_url
+
+    try:
+        given_provider = IdentityProvider[params["provider"].upper()]
+    except KeyError as err:
+        raise httpexceptions.HTTPNotFound from err
+
+    given_provider_unique_id = params["provider_unique_id"]
+
+    identities = db.scalars(select(UserIdentity).where(UserIdentity.user_id == user.id))
+
+    matching_identity = None
+    other_identities = []
+    for identity in identities:
+        if (
+            identity.provider == given_provider
+            and identity.provider_unique_id == given_provider_unique_id
+        ):
+            assert matching_identity is None  # noqa:S101
+            matching_identity = identity
+        else:
+            other_identities.append(identity)
+
+    if not matching_identity:
+        raise httpexceptions.HTTPNotFound
```

Comment:
Reaching this branch is expected to be very rare, though it leads to the standard 404 error page. You could avoid that by treating the operation as a success if the provider was already disconnected, or flash an error for a slightly nicer experience.